fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -1453,15 +1480,20 @@ boolean isErased() {
         public void defineProtoType(final TypeRepository.Builder typeRepositoryBuilder) {
             Objects.requireNonNull(elementType);
             final var typeName = uniqueCompliantTypeName();
-            final var helperDescriptorBuilder = DescriptorProto.newBuilder();
-            helperDescriptorBuilder.setName(typeName);
-            elementType.addProtoField(typeRepositoryBuilder,
-                    helperDescriptorBuilder,
-                    1, "value",
-                    typeRepositoryBuilder.defineAndResolveType(elementType),
-                    FieldDescriptorProto.Label.LABEL_OPTIONAL);
-            typeRepositoryBuilder.addMessageType(helperDescriptorBuilder.build());
             typeRepositoryBuilder.registerTypeToTypeNameMapping(this, typeName);
+            if (needsWrapper && elementType.getTypeCode() != TypeCode.UNKNOWN) {
+                Type wrapperType = Record.fromFields(List.of(Record.Field.of(new Array(elementType), Optional.of(NullableArrayTypeUtils.getRepeatedFieldName()))));
+                typeRepositoryBuilder.defineAndResolveType(wrapperType);
+            } else {
+                final var helperDescriptorBuilder = DescriptorProto.newBuilder();
+                helperDescriptorBuilder.setName(typeName);
+                elementType.addProtoField(typeRepositoryBuilder,
+                        helperDescriptorBuilder,
+                        1, "value",
+                        typeRepositoryBuilder.defineAndResolveType(elementType),
+                        FieldDescriptorProto.Label.LABEL_OPTIONAL);
+                typeRepositoryBuilder.addMessageType(helperDescriptorBuilder.build());
```

Comment:
This is the original logic, I actually don't quite understand it.

Reply:
Remember that the old logic attempts to support arrays that can have `null`s in them. The new logic supports nullable arrays which cannot have `null`s in them. The approaches of how to get that done are eerily similar but not quite the same. I am ok with removing that code that supports the old logic or to completely leave it intact, but please no mix of the two.

Reply:
makes sense. I'll remove this part.